From d879cad155f098492d6c21207226093e3c761772 Mon Sep 17 00:00:00 2001
From: Thao Luong <thao.luong.uw@rvc.renesas.com>
Date: Wed, 17 Jun 2015 19:55:54 +0700
Subject: [PATCH 1/1] omxh264dec: Correct size of buffer when malloc

When retrieve sps and pps to convert avc format to byte stream There are some
malloc with bad size, it can affect when change processor architecture
(32bit -> 64bit). This patch file will correct size when malloc

Signed-off-by: Thao Luong <thao.luong.uw@rvc.renesas.com>
---
 omx/gstomxh264dec.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/omx/gstomxh264dec.c b/omx/gstomxh264dec.c
index 651816f..1995790 100644
--- a/omx/gstomxh264dec.c
+++ b/omx/gstomxh264dec.c
@@ -109,7 +109,7 @@ gst_omx_h264_dec_retrieve_sps_pps (GstOMXH264Dec * self, guint8 * data)
 
   sps_num = ptr[5] & 0x1f;      /* reserved(3bit) + numOfSequenceParameterSets(uint 5bit) */
 
-  sps_size_list = g_malloc (sps_num);
+  sps_size_list = g_malloc (sps_num * sizeof(guint));
   if (!sps_size_list) {
     GST_ERROR_OBJECT (self, "failed g_malloc");
     return NULL;
@@ -124,7 +124,7 @@ gst_omx_h264_dec_retrieve_sps_pps (GstOMXH264Dec * self, guint8 * data)
   }
 
   pps_num = *ptr++;             /* numOfPictureParameterSets (unint 8bit) */
-  pps_size_list = g_malloc (pps_num);
+  pps_size_list = g_malloc (pps_num * sizeof(guint));
   if (!pps_size_list) {
     GST_ERROR_OBJECT (self, "failed g_malloc");
     g_free (sps_size_list);
@@ -137,7 +137,7 @@ gst_omx_h264_dec_retrieve_sps_pps (GstOMXH264Dec * self, guint8 * data)
     sps_pps_size += pps_size_list[i] + 4;       /* take account of the start code length */
   }
 
-  outbuf = g_malloc (sps_pps_size);
+  outbuf = g_malloc (sps_pps_size * sizeof(guint8));
   if (!outbuf) {
     GST_ERROR_OBJECT (self, "failed g_malloc");
     g_free (sps_size_list);
-- 
2.4.0

